source/common/upstream/upstream_impl.cc

```diff
@@ -763,6 +763,16 @@ void ClusterInfoImpl::createNetworkFilterChain(Network::Connection& connection)
   }
 }
 
+Http::Protocol
+ClusterInfoImpl::upstreamHttpProtocol(absl::optional<Http::Protocol> downstream_protocol) const {
+  if (features_ & Upstream::ClusterInfo::Features::USE_DOWNSTREAM_PROTOCOL) {
+    return downstream_protocol.value();
```

Comment:
Is it necessary to check `has_value` here?

Reply:
I don't think so, it wasn't checked when this logic was in router.